src/subtype.c

```diff
@@ -959,12 +959,18 @@ static int subtype_tuple_varargs(struct subtype_tuple_env *env, jl_stenv_t *e, i
             return 0;
         }
 
-        if (ylv && (ylv->depth0 != e->invdepth || ylv->occurs_inv))
-            return 0;
-        e->Rinvdepth++;
-        int ans = subtype(jl_any_type, yp1, e, 2);
-        e->Rinvdepth--;
-        return ans;
+        if (ylv) {
+            if (ylv->depth0 != e->invdepth || ylv->occurs_inv)
+                return 0;
+            e->Rinvdepth++;
+            record_var_occurrence(ylv, e, 2);
+            e->Rinvdepth--;
+        }
+        // We need some sort of sentinel here for the lower bound.
+        // When we still had typevars in every Vararg, this was always
+        // a typevar, but now we just use the vararg marker itself as the
+        // sentinel.
+        return var_gt((jl_tvar_t*)yp1, env->vtx, e, 2);
```

Comment:
Is this passing a VarargMarker to `var_gt`? I think only types should be passed to it. If the only goal is to set the lower bound, I think it's better to do that directly.

Reply:
Oops, guess I shouldn't comment on intermediate commits :)

Reply:
Yes. This doesn't work. I fixed it in a follow up commit.